commands/deployment_strategy.go

```diff
@@ -0,0 +1,46 @@
+package commands
+
+import (
+	"github.com/cloudfoundry-incubator/multiapps-cli-plugin/util"
+	"strconv"
+)
+
+type DeploymentStrategy interface {
+	CreateProcessBuilder() *util.ProcessBuilder
+}
+
+type DeployCommandDeploymentStrategy struct {
+}
+
+func (d *DeployCommandDeploymentStrategy) CreateProcessBuilder() *util.ProcessBuilder {
+	processBuilder := util.NewProcessBuilder()
+	processBuilder.ProcessType(DeployCommandProcessTypeProvider{}.GetProcessType())
```

Comment:
This is not okay from my point of view. Exposing the DeployCommandProcessTypeProvider and the BlueGreenDeployCommandProcessTypeProvider in order to get the process type? Why not the struct contains the process type and in the factory method to accept it??